src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+        assertNull(testUser.getActiveGame());
     }
 
+    @Test
+    void updateActiveGameIfNecessary_newGame() {
+        testUser.setActiveGame(testGame);
+
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        userService.updateActiveGameIfNecessary(testUser.getId());
+        assertEquals(testGame, testUser.getActiveGame());
+    }
+
+    @Test
+    void setMatch(){
+      Match testMatch =  userService.createMatch(testUser, otherUser);
+      testMatch.setMatchId(2L);
+
+      //both users should now have the testmatch
+      userService.setMatch(testMatch);
+
+      //checking if both users have the match with the given id
+      assertTrue(testUser.getMatches().contains(testMatch.getMatchId()));
+        assertTrue(otherUser.getMatches().contains(testMatch.getMatchId()));
+    }
+
+    @Test
+    void otherUserLikesUser_true(){
+      //set the otheruser to like testUser
+      userService.setUserLikesOtherUser(otherUser, testUser);
+      assertTrue(userService.otherUserLikesUser(testUser, otherUser));
+    }
+
+    @Test
+    void otherUserLikesUser_false(){
+        //testUser by default isn't liked by other user
+        assertFalse(userService.otherUserLikesUser(testUser, otherUser));
+    }
+
+    @Test
+    void isGameBelongingToUSer_true(){
+      //by default we set testgame to belong to user
+        testGame.setUserId(1L);
+      assertTrue(userService.isGameBelongingToUser(testGame, testUser));
+    }
+
+    @Test
+    void isGameBelongingToUSer_false(){
+        //we set the testgame to belong to other user by changing the userId of the testgame
+        testGame.setUserId(2L);
+        assertFalse(userService.isGameBelongingToUser(testGame, testUser));
+    }
+
+    @Test
+    void isWhiteCardBelongingToUser_true(){
+      //assign the white cards to the user
+      testUser.setUserWhiteCards(testWhiteCards);
+      long id = testUser.getId();
+
+      Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+      assertTrue(userService.isWhiteCardBelongingToUser(testWhiteCard, testUser.getId()));
+    }
+
+    @Test
+    void isWhiteCardBelongingToUser_false(){
+        //assign the white cards to the otherUser
+        otherUser.setUserWhiteCards(testWhiteCards);
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+        assertFalse(userService.isWhiteCardBelongingToUser(testWhiteCard, testUser.getId()));
+    }
+
+    @Test
+    void deleteWhiteCard(){
+      //Create a second white card to delete
+        WhiteCard deleteCard = new WhiteCard();
+        deleteCard.setId(1L);
+        deleteCard.setText("CardText");
+        testWhiteCards.add(deleteCard); //Add the card to delete to the whitecards for the user
+        testUser.setUserWhiteCards(testWhiteCards);
+
+        long id = testUser.getId();
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
```

Comment:
You could optionally test if the card is belonging to the user here (assertTrue(uS.iWCBTU(deleteCard, testUser.getId()))).

Reply:
done